compiler/next/include/chpl/AST/FnCallExpr.h

```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright 2021 Hewlett Packard Enterprise Development LP
+ * Other additional copyright holders may be indicated within.
+ *
+ * The entirety of this work is licensed under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ *
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef CHPL_AST_FNCALLEXPR_H
+#define CHPL_AST_FNCALLEXPR_H
+
+#include "chpl/AST/CallExpr.h"
+
+#include <vector>
+
+namespace chpl {
+namespace ast {
+
+/**
+  This class represents a function call or method call, including virtual and
+  non-virtual calls (since the difference is not known before resolution).
+ */
+class FnCallExpr final : public CallExpr {
+ private:
+  // this represents the called expression e.g. 'f' in 'f(1,2,3)'
+  Expr* baseExpr_;
```

Comment:
This does not sound right. It should be `child(0)`. `baseExpr()` should not shadow the one in `CallExpr`.

Naming and overriding aside, all child expressions are supposed to go into BaseAST::children_, and this one violates that.

I will write some more under `CallExpr.h` .

Reply:
Yes, I'll fix that, but at the moment, this AST class is totally unused.

Reply:
(I fixed it in a branch for a follow-up PR).